src/bin/help.ts

```diff
@@ -0,0 +1,230 @@
+        PRs match an expected format.
+      `,
+			`
+      --exclude-contributors: Don't add all-contributors to track contributions 
+        and display them in a README.md table.
+      `,
+			`
+      --exclude-lint-json: Don't apply linting and sorting to ${chalk.cyanBright(
+				"*.json",
+			)} and 
+        ${chalk.cyanBright("*.jsonc")} files.
+      `,
+			`
+      --exclude-lint-knip: Don't add Knip to detect unused files, dependencies, 
+        and code exports.
+      `,
+			`
+      --exclude-lint-md: Don't apply linting to ${chalk.cyanBright(
+				"*.md",
+			)} files.
+      `,
+			`
+      --exclude-lint-package-json: Don't add npm-package-json-lint to lint for 
+        package.json correctness.
+      `,
+			`
+      --exclude-lint-packages: Don't add a pnpm dedupe workflow to ensure 
+        packages aren't duplicated unnecessarily.
+      `,
+			`
+      --exclude-lint-perfectionist: Don't apply eslint-plugin-perfectionist to 
+        ensure imports, keys, and so on are in sorted order.
+      `,
+			`
+      --exclude-lint-spelling: Don't add cspell to spell check against 
+        dictionaries of known words.
+      `,
+			`
+      --exclude-lint-yml: Don't apply linting and sorting to ${chalk.cyanBright(
+				"*.yaml",
+			)} and ${chalk.cyanBright("*.yml")} 
+        files.
+      `,
+			`
+      --exclude-releases: Don't add release-it to generate changelogs, package 
+        bumps, and publishes based on conventional commits.
+      `,
+			`
+      --exclude-renovate: Don't add a Renovate config to dependencies 
+        up-to-date with PRs.
+      `,
+			`
+      --exclude-tests: Don't add Vitest tooling for fast unit tests, configured 
+        with coverage tracking.
+      `,
+		),
+	);
+
+	logLine();
+
+	logLine(
+		chalk.green(
+			`
+      You can prevent the migration script from making some network-based 
+      changes using any or all of the following CLI flags:
+      `,
+		),
+	);
+
+	logLine(
+		chalk.cyan(
+			`
+      --exclude-contributors: Skips network calls that fetch 
+        all-contributors data from GitHub
+      `,
+			`
+      --skip-github-api: Skips calling to GitHub APIs.
+      `,
+			`
+      --skip-install: Skips installing all the new template 
+        packages with pnpm.
+      `,
+		),
+	);
+
+	logLine();
+
+	logLine(
+		chalk.green(
+			`
+      You can prevent the migration script from making some changes on disk 
+      using any or all of the following CLI flags:
+      `,
+		),
+	);
+
+	logLine(
+		chalk.cyan(
+			`
+      --skip-removal: Skips removing setup docs and scripts, 
+        including this ${chalk.cyanBright("docs/")} directory
+      `,
+			`
+      --skip-restore: Skips the prompt offering to restore the 
+        repository if an error occurs during setup
+      `,
+			`
+      --skip-uninstall: Skips uninstalling packages only used for 
+        setup scripts
+      `,
+		),
+	);
```

Comment:
Thinking broadly on the output, it looks roughly like a series of repeated tasks:

* Five sections, each of which has a list of options and an optional warning
* Each option has a name, a description, and a type that's either a primitive (`boolean`/`string`) or a union of strings

Both smells strongly to me of a chance to write a standalone function to automate the behavior. That way they're all consistent. Thoughts?

Reply:
Thank you! I've gone for this approach now and I think its a lot cleaner, and it dovetails nicely with the merging of runtime areas you mentioned below as well.